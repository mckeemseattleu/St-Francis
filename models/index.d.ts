import { Timestamp } from 'firebase/firestore';

export declare type Client = {
    readonly id: string;
    readonly notes?: string | null;
    readonly gender?: string | null;
    readonly race?: string | null;
    readonly numKids?: number | null;
    readonly birthday?: Timestamp | null;
    readonly isBanned?: boolean | null;
    readonly lastName?: string | null;
    readonly unhoused?: boolean | null;
    readonly createdAt?: Timestamp | null;
    readonly firstName?: string | null;
    readonly lastVisit?: Timestamp | null;
    readonly sheltered?: boolean | null;
    readonly updatedAt?: Timestamp | null;
    readonly postalCode?: string | null;
    readonly isCheckedIn?: boolean | null;
    readonly BPCResident?: boolean | null;
    readonly middleInitial?: string | null;
    readonly lastBackpack?: Timestamp | null;
    readonly lastBusTicket?: Timestamp | null;
    readonly lastNameLower?: string | null;
    readonly firstNameLower?: string | null;
    readonly lastSleepingBag?: Timestamp | null;
    readonly idv1?: string;
};

export declare type Visit = {
    readonly id: string;
    readonly mensQ?: number | null;
    readonly kidsQ?: number | null;
    readonly notes?: string | null;
    readonly boyAge?: string | null;
    readonly diaper?: number | null;
    readonly girlAge?: string | null;
    readonly womensQ?: number | null;
    readonly backpack?: boolean | null;
    readonly giftCard?: number | null;
    readonly busTicket?: number | null;
    readonly orcaCard?: number | null;
    readonly household?: string | null;
    readonly createdAt?: Timestamp | null;
    readonly clothingBoy?: boolean | null;
    readonly clothingMen?: boolean | null;
    readonly clothingOther?: boolean | null;
    readonly otherQ?: number | null;
    readonly sleepingBag?: boolean | null;
    readonly food?: boolean | null;
    readonly clothingGirl?: boolean | null;
    readonly clothingWomen?: boolean | null;
    readonly financialAssistance?: number | null;
    readonly idv1?: string;
    readonly householdItem?: boolean | null;
    readonly householdItemQ?: number | null;
    readonly clothingKids?: boolean | null;
};

export declare type Settings = {
    readonly id: string;
    readonly daysEarlyThreshold?: number;
    readonly backpackThreshold?: number;
    readonly sleepingBagThreshold?: number;
    readonly earlyOverride?: boolean;
    readonly createdAt?: Timestamp | null;
    readonly updatedAt?: Timestamp | null;
};
