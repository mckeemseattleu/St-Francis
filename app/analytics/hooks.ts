import { Client } from '@/models/index';
import { CLIENTS_PATH } from '@/utils/constants';
import { DocFilter } from '@/utils/fetchData';
import { listClients } from '@/utils/queries';
import { useEffect, useState } from 'react';
import { useQuery } from 'react-query';

const MAX_CLIENTS_PER_REQUEST = 9999;

export const useGetDashboardData = () => {
    const [fields, setFields] = useState<DocFilter>();

    const { data, isLoading, refetch, error } = useQuery({
        queryKey: [CLIENTS_PATH, 'report'],
        queryFn: async () => {
            let clients: Array<Client> = [];
            clients = await listClients(fields, MAX_CLIENTS_PER_REQUEST);
            return { clients, fields };
        },
        enabled: false,
    });

    useEffect(() => {
        if (fields) refetch();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [fields]);

    return { data, isLoading, refetch, error, setFields };
};
