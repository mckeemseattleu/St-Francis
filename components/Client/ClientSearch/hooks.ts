import { useMutation } from 'react-query';
import { useState } from 'react';
import { subDays } from 'date-fns';
import useAlert from '@/hooks/useAlert';
import type { Client } from '@/models/index';
import { DocFilter, FilterObject, listVisits } from '@/utils/index';
import { listClients, getSettings } from '@/utils/index';
import type { VisitWithClientId } from '@/types/visits';

export const useGetClientsSearch = () => {
    const [, setAlert] = useAlert();
    const [clients, setClients] = useState<Client[]>([]);
    const [visits, setVisits] = useState<VisitWithClientId[]>([]);

    const { mutateAsync, isLoading } = useMutation({
        mutationFn: async (fields: DocFilter) => {
            const filter = { ...fields };
            if (filter?.filterByBirthday && !filter.birthday) {
                setAlert({
                    message: 'invalid date of birth',
                    type: 'error',
                });
            }
            if (filter?.filterByBirthday && filter?.birthday) {
                filter.birthday = new Date(filter.birthday as string);
                delete filter.filterByBirthday;
            }
            if (!Object.keys(filter).length)
                filter.updatedAt = {
                    opStr: '>=',
                    value: new Date(new Date().toDateString()),
                } as FilterObject;
            const clients = await listClients(filter);

            // fetch visit by clients
            const userIds = clients.map((client) => client.id);
            const visits = await getVisitsByClientIds(userIds);
            setVisits(visits);
            setClients(clients);
        },
    });

    return { mutateAsync, isLoading, clients, visits, setClients };
};

const VISITS_LIMIT = 10; //  visit limit per client

export const getVisitsByClientIds = async (userIds: string[]) => {
    const ORCA_CARD_TRESHOLD = (await getSettings()).orcaCardThreshold;
    if (!ORCA_CARD_TRESHOLD) {
        return [];
    }

    const pastDate = subDays(new Date(), ORCA_CARD_TRESHOLD);

    const promises = userIds.map(async (userId) => {
        try {
            const visits = await listVisits(
                userId,
                {
                    createdAt: { opStr: '>=', value: pastDate },
                },
                VISITS_LIMIT
            );

            // Add the clientId to each visit
            return visits.map((visit) => ({ ...visit, clientId: userId }));
        } catch (error) {
            return [];
        }
    });

    // Resolve all promises and flatten the results
    const results = await Promise.all(promises);
    const visits = results.flat();

    return visits;
};
