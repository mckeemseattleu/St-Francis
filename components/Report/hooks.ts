import { useMutation } from 'react-query';
import type { ReportDateRange } from './types';
import {
    getClients,
    getClientsVisits,
    generateVisitData,
    generateClientData,
    generateCSV,
} from './utils';

export const useGenerateReport = () => {
    const { mutateAsync, isError, isLoading } = useMutation({
        mutationFn: async (reportDateRange: ReportDateRange) => {
            // get data from firebase
            const listClients = await getClients(reportDateRange);
            const clientsVisits = await getClientsVisits(
                listClients,
                reportDateRange
            );

            // generate data by clients and by products
            const clientsData = generateClientData(listClients);
            const reportVisits = generateVisitData(clientsVisits);

            // generate csv
            generateCSV(clientsData, reportVisits, reportDateRange);
        },
    });
    return { mutateAsync, isError, isLoading };
};
