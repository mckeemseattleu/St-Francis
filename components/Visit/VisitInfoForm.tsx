'use client';

import { Button, Form, FormItem, FormRow } from '@/components/UI';
import { Visit } from '@/models/index';
import clsx from 'clsx';
import { useState } from 'react';
import styles from './VisitInfoForm.module.css';

interface VisitInfoFormProps {
    initialVisitData?: Visit;
    onSubmit?: (visitData: Visit) => void;
    submitLabel?: string;
    onChange?: (visitData: Visit) => void;
    isCheckIn?: boolean;
}

const defaultVisitData = {
    id: '',
    clothingMen: false,
    clothingWomen: false,
    clothingBoy: false,
    clothingGirl: false,
    household: '',
    notes: '',
    backpack: false,
    sleepingBag: false,
    food: false,
    mensQ: '',
    womensQ: '',
    kidsQ: '',
    householdItem: false,
    householdItemQ: '',
    clothingKids: false,
};

const toInt = (value: string) => parseInt(value) || 0;
const toString = (value: number | undefined | null) => {
    if (value === 0) return '';
    return value?.toString() || '';
};

export default function VisitInfoForm({
    onSubmit,
    initialVisitData,
    submitLabel,
    onChange,
    isCheckIn,
}: VisitInfoFormProps) {
    // Append boyAge and girlAge to notes
    let initialNotes = initialVisitData?.notes || '';
    initialVisitData?.boyAge &&
        (initialNotes += `\nBoy Age: ${initialVisitData?.boyAge} `);
    initialVisitData?.girlAge &&
        (initialNotes += `\nGirl Age: ${initialVisitData?.girlAge} `);

    const [visitData, setVisitData] = useState({
        ...defaultVisitData,
        ...initialVisitData,
        busTicket: toString(initialVisitData?.busTicket),
        orcaCard: toString(initialVisitData?.orcaCard),
        giftCard: toString(initialVisitData?.giftCard),
        diaper: toString(initialVisitData?.diaper),
        financialAssistance: toString(initialVisitData?.financialAssistance),
        mensQ: toString(initialVisitData?.mensQ),
        womensQ: toString(initialVisitData?.womensQ),
        kidsQ: toString(initialVisitData?.kidsQ),
        notes: initialNotes,
        householdItem:
            initialVisitData?.householdItem || initialVisitData?.household
                ? true
                : false,
        householdItemQ: toString(initialVisitData?.householdItemQ),

        // carry existed boy & girl clothing state to clothingKids
        clothingKids:
            initialVisitData?.clothingBoy ||
            initialVisitData?.clothingGirl ||
            initialVisitData?.clothingKids
                ? true
                : false,
    });

    const handleSubmit = async (e: any) => {
        e.preventDefault(); // Prevent redirect
        onSubmit && onSubmit(transformData(visitData));
    };

    const transformData = (visitData: any) => {
        if (isCheckIn) {
            // During check-in, don't set any numerical values
            return {
                ...visitData,
                orcaCard: visitData.orcaCard ? true : null,
                giftCard: visitData.giftCard ? true : null,
                diaper: visitData.diaper ? true : null,
                // ... other transformations
            };
        }
        
        // During checkout, handle the actual values
        return {
            ...visitData,
            orcaCard: toInt(visitData.orcaCard),
            giftCard: toInt(visitData.giftCard),
            diaper: toInt(visitData.diaper),
            // ... other transformations
        };
    };

    const handleChange = (key: string) => (e: any) => {
        let value: string | number | boolean = '';
        
        if (isCheckIn && ['orcaCard', 'giftCard', 'diaper'].includes(key)) {
            value = e.target.checked;
        } else {
            // Normal handling for other fields and checkout mode
            if (e.target.type === 'checkbox') {
                value = e.target.checked;
            } else if (e.target.type === 'number') {
                value = e.target.value;
            } else {
                value = e.target.value;
            }
        }
        
        const data = transformData({ ...visitData, [key]: value });
        onChange && onChange(data);
        setVisitData({ ...visitData, [key]: value });
    };
    return (
        <Form onSubmit={handleSubmit}>
            <h2>Clothing</h2>

            <FormRow className={styles.formRows}>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        label="Men"
                        type="checkbox"
                        id="clothingMen"
                        checked={!!visitData.clothingMen}
                        onChange={handleChange('clothingMen')}
                    />
                    <FormItem
                        type="number"
                        id="mensQ"
                        placeholder="Count"
                        value={visitData.mensQ}
                        onChange={handleChange('mensQ')}
                        hidden={!visitData.clothingMen}
                    />
                </FormRow>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        type="checkbox"
                        id="clothingWomen"
                        label="Women"
                        checked={!!visitData.clothingWomen}
                        onChange={handleChange('clothingWomen')}
                    />
                    <FormItem
                        type="number"
                        id="womensQ"
                        placeholder="Count"
                        value={visitData.womensQ}
                        onChange={handleChange('womensQ')}
                        hidden={!visitData.clothingWomen}
                    />
                </FormRow>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        type="checkbox"
                        id="clothingKids"
                        label="Kids"
                        checked={!!visitData.clothingKids}
                        onChange={handleChange('clothingKids')}
                    />
                    <FormItem
                        type="number"
                        id="kidsQ"
                        placeholder="Count"
                        value={visitData.kidsQ}
                        onChange={handleChange('kidsQ')}
                        hidden={
                            !visitData.clothingBoy &&
                            !visitData.clothingGirl &&
                            !visitData.clothingKids
                        }
                    />
                </FormRow>
            </FormRow>

            <h2>Special Requests</h2>

            <FormRow>
                <FormItem
                    type="number"
                    id="busTicket"
                    label="Bus Ticket"
                    placeholder="Count"
                    value={visitData.busTicket}
                    onChange={handleChange('busTicket')}
                />
                {isCheckIn ? (
                    // Show checkboxes during check-in
                    <>
                        <FormItem
                            type="checkbox"
                            id="orcaCard"
                            label="Orca Card"
                            checked={!!visitData.orcaCard}
                            onChange={handleChange('orcaCard')}
                        />
                        <FormItem
                            type="checkbox"
                            id="giftCard"
                            label="Gift Card"
                            checked={!!visitData.giftCard}
                            onChange={handleChange('giftCard')}
                        />
                        <FormItem
                            type="checkbox"
                            id="diaper"
                            label="Diaper"
                            checked={!!visitData.diaper}
                            onChange={handleChange('diaper')}
                        />
                    </>
                ) : (
                    // Show number inputs during checkout
                    <>
                        <FormItem
                            type="number"
                            id="orcaCard"
                            placeholder="Value in Dollars"
                            label="Orca Card"
                            value={visitData.orcaCard}
                            onChange={handleChange('orcaCard')}
                        />
                        <FormItem
                            type="number"
                            id="giftCard"
                            label="Gift Card"
                            placeholder="Value in Dollars"
                            value={visitData.giftCard}
                            onChange={handleChange('giftCard')}
                        />
                        <FormItem
                            type="number"
                            id="diaper"
                            label="Diaper"
                            placeholder="Diaper Count"
                            value={visitData.diaper}
                            onChange={handleChange('diaper')}
                        />
                    </>
                )}
                <FormItem
                    type="number"
                    id="financialAssistance"
                    label="Financial Assistance"
                    placeholder="Value in Dollars"
                    value={visitData.financialAssistance}
                    onChange={handleChange('financialAssistance')}
                />
            </FormRow>
            <FormRow>
                <FormItem
                    type="checkbox"
                    id="backpack"
                    label="Backpack"
                    checked={!!visitData.backpack}
                    onChange={handleChange('backpack')}
                />
                <FormItem
                    type="checkbox"
                    id="sleepingBag"
                    label="Sleeping Bag"
                    checked={!!visitData.sleepingBag}
                    onChange={handleChange('sleepingBag')}
                />
                <FormItem
                    type="checkbox"
                    id="food"
                    label="Food"
                    checked={!!visitData.food}
                    onChange={handleChange('food')}
                />
                <div className={clsx(styles.rowItems, styles.householdRow)}>
                    <FormItem
                        type="checkbox"
                        id="householdItem"
                        label="Household Items"
                        checked={!!visitData.householdItem}
                        onChange={handleChange('householdItem')}
                    />

                    <FormItem
                        type="number"
                        id="householdItemQ"
                        placeholder="Count"
                        value={visitData.householdItemQ}
                        onChange={handleChange('householdItemQ')}
                        hidden={!visitData.householdItem}
                    />
                </div>
            </FormRow>

            <div hidden={!visitData.householdItem}>
                <h2>Household items</h2>
                <FormItem
                    type="textarea"
                    id="household"
                    rows={3}
                    value={visitData.household || ''}
                    onChange={handleChange('household')}
                />
            </div>

            <h2>Notes</h2>

            <FormItem
                type="textarea"
                id="notes"
                rows={3}
                value={visitData.notes || ''}
                onChange={handleChange('notes')}
            />
            <Button type="submit" className={styles.submitBtn}>
                {submitLabel || 'Submit'}
            </Button>
        </Form>
    );
}
