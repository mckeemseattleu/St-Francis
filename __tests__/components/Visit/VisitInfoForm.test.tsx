import VisitInfoForm from '@/components/Visit/VisitInfoForm';
import { Client } from '@/models/index';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Timestamp } from 'firebase/firestore';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { useSettings } from '@/hooks/index';

jest.mock('@/hooks/index', () => ({
    useSettings: jest.fn(() => ({
        settings: {
            visitThreshold: 30,
            backpackThreshold: 180,
            sleepingBagThreshold: 180,
        },
    })),
}));

describe('Visit Info Form Component', () => {
    const getInputField = (label: string) => {
        const input = screen.getByLabelText(label) as HTMLInputElement;
        if (input.type === 'checkbox') return input.checked;
        return input.value;
    };
    const mockVisitData = {
        id: 'mockVisitId',
        clothingMen: true,
        clothingWomen: true,
        clothingBoy: true,
        clothingGirl: true,
        backpack: true,
        sleepingBag: true,
        busTicket: 1,
        giftCard: 2,
        diaper: 3,
        financialAssistance: 4,
    };

    it('matches snapshot without any props', async () => {
        const tree = renderer.create(<VisitInfoForm />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without initial data', async () => {
        render(<VisitInfoForm />);
        expect(screen.queryByText('Men')).toBeInTheDocument();
        expect(screen.queryByText('Women')).toBeInTheDocument();
        expect(screen.queryByText('Kids')).toBeInTheDocument();
        expect(screen.queryByText('Backpack')).toBeInTheDocument();
        expect(screen.queryByText('Sleeping Bag')).toBeInTheDocument();
        expect(screen.queryByText('Bus Ticket')).toBeInTheDocument();
        expect(screen.queryByText('Gift Card')).toBeInTheDocument();
        expect(screen.queryByText('Diaper')).toBeInTheDocument();
        expect(screen.queryByText('Financial Assistance')).toBeInTheDocument();
    });

    it('renders correctly with initial Visit Data', async () => {
        render(<VisitInfoForm initialVisitData={mockVisitData} />);
        expect(getInputField('Men')).toBe(true);
        expect(getInputField('Women')).toBe(true);
        expect(getInputField('Kids')).toBe(true);
        expect(getInputField('Backpack')).toBe(true);
        expect(getInputField('Sleeping Bag')).toBe(true);
        expect(getInputField('Bus Ticket')).toBe('1');
        expect(getInputField('Gift Card')).toBe('2');
        expect(getInputField('Diaper')).toBe('3');
        expect(getInputField('Financial Assistance')).toBe('4');
    });

    it('saves correct output with mock onSubmit callback', async () => {
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(<VisitInfoForm onSubmit={mockOnSave} />);
        const saveButton = screen.getAllByRole('button', {
            name: 'Submit',
        })[0];
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });

    it('changes input value correctly', async () => {
        render(<VisitInfoForm />);
        const textboxes = screen.getAllByRole('textbox');
        const input = textboxes[0] as HTMLInputElement;
        await userEvent.type(input, '123');
        expect(input.value).toBe('123');
    });

    it('changes input value correctly', async () => {
        render(<VisitInfoForm />);
        const input = screen.getByRole('spinbutton', {
            name: /bus ticket/i,
        }) as HTMLInputElement;
        await userEvent.clear(input);
        const saveButton = screen.getAllByRole('button', {
            name: 'Submit',
        })[0];
        fireEvent.click(saveButton);
    });

    it('allows early override to checkin', async () => {
        (useSettings as jest.Mock).mockReturnValue({
            settings: { earlyOverride: true },
        });
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(<VisitInfoForm onSubmit={mockOnSave} />);
        const saveButton = screen.getAllByRole('button', {
            name: 'Submit',
        })[0];
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });
});
