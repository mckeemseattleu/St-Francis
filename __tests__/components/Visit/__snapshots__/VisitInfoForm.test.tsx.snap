// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visit Info Form Component matches snapshot without any props 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Clothing
  </h2>
  <div
    className="formRows"
  >
    <div
      className="rowItems"
    >
      <div
        className="formItem"
      >
        <label
          htmlFor="clothingMen"
        >
          Men
        </label>
        <input
          checked={false}
          id="clothingMen"
          label="Men"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="formItem"
      >
        <label
          htmlFor="mensQ"
        />
        <input
          hidden={true}
          id="mensQ"
          min={0}
          onChange={[Function]}
          placeholder="Count"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="rowItems"
    >
      <div
        className="formItem"
      >
        <label
          htmlFor="clothingWomen"
        >
          Women
        </label>
        <input
          checked={false}
          id="clothingWomen"
          label="Women"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="formItem"
      >
        <label
          htmlFor="womensQ"
        />
        <input
          hidden={true}
          id="womensQ"
          min={0}
          onChange={[Function]}
          placeholder="Count"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="rowItems"
    >
      <div
        className="formItem"
      >
        <label
          htmlFor="clothingKids"
        >
          Kids
        </label>
        <input
          checked={false}
          id="clothingKids"
          label="Kids"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="formItem"
      >
        <label
          htmlFor="kidsQ"
        />
        <input
          hidden={true}
          id="kidsQ"
          min={0}
          onChange={[Function]}
          placeholder="Count"
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
  <h2>
    Special Requests
  </h2>
  <div
    className="formRow"
  >
    <div
      className="formItem"
    >
      <label
        htmlFor="busTicket"
      >
        Bus Ticket
      </label>
      <input
        id="busTicket"
        label="Bus Ticket"
        min={0}
        onChange={[Function]}
        placeholder="Count"
        type="number"
        value=""
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="orcaCard"
      >
        Orca Card
      </label>
      <input
        id="orcaCard"
        label="Orca Card"
        min={0}
        onChange={[Function]}
        placeholder="Count"
        type="number"
        value=""
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="giftCard"
      >
        Gift Card
      </label>
      <input
        id="giftCard"
        label="Gift Card"
        min={0}
        onChange={[Function]}
        placeholder="Value in Dollars"
        type="number"
        value=""
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="diaper"
      >
        Diaper
      </label>
      <input
        id="diaper"
        label="Diaper"
        min={0}
        onChange={[Function]}
        placeholder="Count"
        type="number"
        value=""
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="financialAssistance"
      >
        Financial Assistance
      </label>
      <input
        id="financialAssistance"
        label="Financial Assistance"
        min={0}
        onChange={[Function]}
        placeholder="Value in Dollars"
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="formRow"
  >
    <div
      className="formItem"
    >
      <label
        htmlFor="backpack"
      >
        Backpack
      </label>
      <input
        checked={false}
        id="backpack"
        label="Backpack"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="sleepingBag"
      >
        Sleeping Bag
      </label>
      <input
        checked={false}
        id="sleepingBag"
        label="Sleeping Bag"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="food"
      >
        Food
      </label>
      <input
        checked={false}
        id="food"
        label="Food"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="rowItems householdRow"
    >
      <div
        className="formItem"
      >
        <label
          htmlFor="householdItem"
        >
          Household Items
        </label>
        <input
          checked={false}
          id="householdItem"
          label="Household Items"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="formItem"
      >
        <label
          htmlFor="householdItemQ"
        />
        <input
          hidden={true}
          id="householdItemQ"
          min={0}
          onChange={[Function]}
          placeholder="Count"
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    hidden={true}
  >
    <h2>
      Household items
    </h2>
    <div
      className="formItem"
    >
      <label
        htmlFor="household"
      />
      <textarea
        id="household"
        onChange={[Function]}
        rows={3}
        type="textarea"
        value=""
      />
    </div>
  </div>
  <h2>
    Notes
  </h2>
  <div
    className="formItem"
  >
    <label
      htmlFor="notes"
    />
    <textarea
      id="notes"
      onChange={[Function]}
      rows={3}
      type="textarea"
      value=""
    />
  </div>
  <button
    className="submitBtn defaultButton"
    type="submit"
  >
    Submit
  </button>
</form>
`;
